src/store/audio.ts

```diff
@@ -1706,14 +1718,16 @@ export const audioStore = createPartialStore<AudioStoreTypes>({
           });
         })();
 
-        const writeFileResult = await window.electron.writeFile({
-          filePath,
-          buffer: await textBlob.arrayBuffer(),
-        });
-        if (writeFileResult) {
-          window.electron.logError(new Error(writeFileResult.message));
-          return { result: "WRITE_ERROR", path: filePath };
-        }
+        await window.electron
+          .writeFile({
+            filePath,
+            buffer: await textBlob.arrayBuffer(),
+          })
+          .then((result) => {
+            if (result.ok) return;
+            window.electron.logError(result.error);
+            return { result: "WRITE_ERROR", path: filePath };
+          });
```

Comment:
これたぶん`result.ok == false`でもpromiseが抜けるだけでエラーがreturnされずに次に進みますね･･･。

こうですかねぇ

```suggestion
        const writeFileResult = await window.electron
          .writeFile({
            filePath,
            buffer: await textBlob.arrayBuffer(),
          })
        if (!writeFileResult.ok) {
            window.electron.logError(new Error(writeFileResult.message));
            return { result: "WRITE_ERROR", path: filePath };
        }
```

Reply:
確かに。
この関数自体try-catchで囲んで、getValueOrThrowしてcatchでWRITE_ERROR返す方が綺麗な気がしました。

Reply:
try-catchで囲みました。

そもそもこの関数をResult型返すようにした方が良いような気がしてます。